PyInstaller/lib/modulegraph/modulegraph.py

```diff
@@ -2721,44 +2773,48 @@ def _process_imports(self, source_module):
 
         # For each target module imported by this source module...
         for have_star, import_info, kwargs in source_module._deferred_imports:
-            # Graph node of the target module specified by the "from" portion
-            # of this "from"-style star import (e.g., an import resembling
-            # "from {target_module_name} import *") or ignored otherwise.
-            target_module = self._safe_import_hook(*import_info, **kwargs)[0]
-
-            # If this is a "from"-style star import, process this import.
-            if have_star:
-                #FIXME: Sadly, the current approach to importing attributes
-                #from "from"-style star imports is... simplistic. This should
-                #be revised as follows. If this target module is:
-                #
-                #* A package:
-                #  * Whose "__init__" submodule defines the "__all__" global
-                #    attribute, only attributes listed by this attribute should
-                #    be imported.
-                #  * Else, *NO* attributes should be imported.
-                #* A non-package:
-                #  * Defining the "__all__" global attribute, only attributes
-                #    listed by this attribute should be imported.
-                #  * Else, only public attributes whose names are *NOT*
-                #    prefixed by "_" should be imported.
-                source_module.add_global_attrs_from_module(target_module)
-
-                source_module._starimported_ignored_module_names.update(
-                    target_module._starimported_ignored_module_names)
-
-                # If this target module has no code object and hence is
-                # unparsable, record its name for posterity.
-                if target_module.code is None:
-                    target_module_name = import_info[0]
-                    source_module._starimported_ignored_module_names.add(
-                        target_module_name)
+            self._process_imports_module(source_module, have_star, import_info,
+                                         kwargs)
 
         # For safety, prevent these imports from being reprocessed.
         source_module._deferred_imports = None
 
-
-    def _load_package(self, fqname, pathname, pkgpath):
+    def _process_imports_module(self, source_module, have_star, import_info,
```

Comment:
Is there anything changed related to the prior code which was part of `_process_imports()`?

Anyway, this new function is used exactly once. So what is the reason for moving the code out of `_process_imports()`? 

Having an extra method raises quite some questions for anybody trying to understand the code: Where is this method defined?` Where else this method used? I'd prefer to not add this additional layer of complexity and to keep the change small.